frame/contracts/src/schedule.rs

```diff
@@ -73,12 +58,31 @@ pub struct Schedule<T: Config> {
 	pub(crate) host_fn_weights: HostFnWeights<T>,
 }
 
+impl<T: Config> Schedule<T> {
+	/// Set the version of the instruction weights.
+	///
+	/// # Note
+	///
+	/// Should be incremented whenever any instruction weight is changed. The
+	/// reason is that changes to instruction weights require a re-instrumentation
+	/// in order to apply the changes to an already deployed code. The re-instrumentation
+	/// is triggered by comparing the version of the current schedule with the the code was
+	/// instrumented with. Changes usually happen when pallet_contracts is re-benchmarked.
+	///
+	/// Changes to other parts of the schedule should not increment the version in
+	/// order to avoid unnecessary re-instrumentations.
+	pub fn set_instruction_weights_version(&mut self, version: u32) {
```

Comment:
Consider a chain that switches its reference machine and therefore re-runs its benchmarks. The code of the `Default` impl stays the same but the values it generates are different because it takes them from the benchmark results. In that case the runtime needs this function in order to trigger a re-instrumentation.

Another use case would be that a chain wants to suppress a re-instrumentation triggered by changes to `InstructionWeights` because it rather suffers slightly wrong weights than a re-instrumentation of all contracts.